### PR TITLE
John conroy/home revision upper grid

### PR DESCRIPTION
**Pull request by john-conroy**
Still need to add opacity transitions for the carousel, the carousel's text, and the bar chart. The carousel's image order is fixed and there is no autoplay, you can only navigate the images through the provided buttons. I'd recommend checking this out locally and also taking a look at various screen sizes.

The library I'm using for the carousel (and I've tried out a couple) doesn't allow for an `onSlideChange` instead only before/after slide callbacks. This results in the text being changed after the slide when the change is initiated by a keydown on one of the arrow keys when the carousel is focused. When the slide is changed by the buttons everything changes at once. I'll look into fixing this when I tackle animations, I might have to just add our own listeners.

Screen Shot 2021-04-14 at 11 15 21 AM
Screen Shot 2021-04-14 at 11 15 12 AM
Screen Shot 2021-04-14 at 11 15 05 AM

### CODE REVIEW

**Review by tsliaw — APPROVED** [review]
Everything looks good to me for all screen sizes. Once you add animation for a smoother transition, the carousel should look great.

**Review by mccalluc — CHANGES_REQUESTED** [review]
- Right now, there's a big gap between the carousel and the bar charts when the screen is wide -- something will fill that gap?
- On narrow screens, the barchart doesn't seem to be on the page at all... intentional?
- If the images go to a submodule, could you add a readme giving the steps with imagemagick (or tool of your choice) to generate the resolutions we need?
- How do things work if the image doesn't have the right aspect ratio? Is that handled in the CSS, or does the processing pipeline need to pad things out so the ratio is consistent?
- The PNGs themselves are larger than they need to be. I just used the first result on google, and got 50% lossless compression. For the command line, probably `pngcrush`. (Assuming now is the time to optimize delivery? It feels like you've gotten this far, we should make sure we have a good pipeline before moving on... but if we hadn't even gotten into image sizing in this PR, that wouldn't have bothered me either ... ie, I think it makes sense to sort this out while we're thinking about it, but I don't think it's a super high priority in its own right, if that makes sense.)

**Comment by john-conroy** [comment]
> Right now, there's a big gap between the carousel and the bar charts when the screen is wide -- something will fill that gap?
> On narrow screens, the barchart doesn't seem to be on the page at all... intentional?

Yes, to both questions. Here's a [link to the design](https://www.figma.com/file/cEZdI2tOMByTBW8RQ2kuAm/Home-Page?node-id=555%3A1).

> How do things work if the image doesn't have the right aspect ratio? Is that handled in the CSS, or does the processing pipeline need to pad things out so the ratio is consistent?

I think for the carousel, we should just only use images with the correct ration (16:9).

> The PNGs themselves are larger than they need to be. I just used the first result on google, and got 50% lossless compression. For the command line, probably pngcrush. (Assuming now is the time to optimize delivery? It feels like you've gotten this far, we should make sure we have a good pipeline before moving on... but if we hadn't even gotten into image sizing in this PR, that wouldn't have bothered me either ... ie, I think it makes sense to sort this out while we're thinking about it, but I don't think it's a super high priority in its own right, if that makes sense.)

I used ImageOptim to minify the images, but I'll try another tool.

File: context/app/static/js/components/home-revision/CarouselImage/Carouselmage.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import { routeContainerMaxWidth, routeContainerPadding } from 'js/components/Routes/Route/style';
+import { callToActionMdOrLargerWidth } from 'js/components/home-revision/ImageCarouselContainer/style';
+import { StyledImage } from './style';
+
+// Responsive images: https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images
+// Setting height and width: https://www.smashingmagazine.com/2020/03/setting-height-width-images-important-again/
+// 1392 was the max resoultion provided by the designer.
+
+function Carouselmage({ src320w, src640w, src800w, src1024w, src1280w, src1392w, alt }) {
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/home-revision/CarouselImage/Carouselmage.jsx:12]: Sure, if you feel like it cuts complexity we can reduce the srcset with probably little noticeable issue.

File: context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    >
+      <CarouselImage
+        src320w={VitessceSlide320w}
+        src640w={VitessceSlide640w}
+        src800w={VitessceSlide800w}
+        src1024w={VitessceSlide1024w}
+        src1280w={VitessceSlide1280w}
+        src1392w={VitessceSlide1392w}
+        alt="Vitessce"
+      />
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx:49]: Yeah, this was messier than I'd prefer. An object with an associated util function could be better. Or we could change how we import images, but that'd require a webpack tweak.

File: context/app/static/js/components/home-revision/ImageCarouselControlButtons/ImageCarouselControlButtons.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  );
+}
+
+function ImageCarouselControlButtons({ numImages, selectedImageIndex, setSelectedImageIndex }) {
+  function setNextSelectedImageIndex() {
+    setSelectedImageIndex(selectedImageIndex === numImages - 1 ? 0 : selectedImageIndex + 1);
+  }
+
+  function setPreviousSelectedImageIndex() {
+    setSelectedImageIndex(selectedImageIndex === 0 ? numImages - 1 : selectedImageIndex - 1);
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/home-revision/ImageCarouselControlButtons/ImageCarouselControlButtons.jsx:26]: Not sure if a mod is much simpler, but sure.

**Comment by john-conroy** [comment]
Filed #1740 and #1741.

File: context/app/static/js/components/home-revision/ImageCarouselControlButtons/ImageCarouselControlButtons.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      ) : (
+        <PanoramaFishEyeRoundedIcon fontSize="small" />
+      )}
+    </StyledIconButton>
+  );
+}
+
+function ImageCarouselControlButtons({ numImages, selectedImageIndex, setSelectedImageIndex }) {
+  function setNextSelectedImageIndex() {
+    setSelectedImageIndex(selectedImageIndex === numImages - 1 ? 0 : selectedImageIndex + 1);
```
Comment (mccalluc) [inline context/app/static/js/components/home-revision/ImageCarouselControlButtons/ImageCarouselControlButtons.jsx:22]:
```suggestion
    setSelectedImageIndex((selectedImageIndex + 1) % numImages);
```

File: context/app/static/js/components/home-revision/ImageCarouselControlButtons/ImageCarouselControlButtons.jsx (hunk 3 above)
Comment (mccalluc) [inline context/app/static/js/components/home-revision/ImageCarouselControlButtons/ImageCarouselControlButtons.jsx:26]:
```suggestion
    setSelectedImageIndex((selectedImageIndex - 1 + numImages) % numImages); // In JS, "%" is remainder, not modulus.
```

File: context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      <CarouselImage
+        src320w={AzimuthSlide320w}
+        src640w={AzimuthSlide640w}
+        src800w={AzimuthSlide800w}
+        src1024w={AzimuthSlide1024w}
+        src1280w={AzimuthSlide1280w}
+        src1392w={AzimuthSlide1392w}
+        alt="Azimuth"
+      />
```
Comment (mccalluc) [inline context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx:67]:
```suggestion
      <CarouselImage
        src320w={CCFSlide320w}
        src640w={CCFSlide640w}
        src1280w={CCFSlide1280w}
        src1392w={CCFSlide1392w}
        alt="CCF Portal"
      />
      <CarouselImage
        src320w={AzimuthSlide320w}
        src640w={AzimuthSlide640w}
        src1280w={AzimuthSlide1280w}
        src1392w={AzimuthSlide1392w}
        alt="Azimuth"
      />
```

File: context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    >
+      <CarouselImage
+        src320w={VitessceSlide320w}
+        src640w={VitessceSlide640w}
+        src800w={VitessceSlide800w}
+        src1024w={VitessceSlide1024w}
+        src1280w={VitessceSlide1280w}
+        src1392w={VitessceSlide1392w}
+        alt="Vitessce"
+      />
```
Comment (mccalluc) [inline context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx:49]:
```suggestion
      <CarouselImage
        src320w={VitessceSlide320w}
        src640w={VitessceSlide640w}
        src1280w={VitessceSlide1280w}
        src1392w={VitessceSlide1392w}
        alt="Vitessce"
      />
```

File: context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import AzimuthSlide800w from 'portal-images/azimuth-slide-800w.png';
+import AzimuthSlide1024w from 'portal-images/azimuth-slide-1024w.png';
+import AzimuthSlide1280w from 'portal-images/azimuth-slide-1280w.png';
+import AzimuthSlide1392w from 'portal-images/azimuth-slide-1392w.png';
+import CCFSlide320w from 'portal-images/ccf-slide-320w.png';
+import CCFSlide640w from 'portal-images/ccf-slide-640w.png';
+import CCFSlide800w from 'portal-images/ccf-slide-800w.png';
+import CCFSlide1024w from 'portal-images/ccf-slide-1024w.png';
+import CCFSlide1280w from 'portal-images/ccf-slide-1280w.png';
+import CCFSlide1392w from 'portal-images/ccf-slide-1392w.png';
```
Comment (mccalluc) [inline context/app/static/js/components/home-revision/ImageCarousel/ImageCarousel.jsx:22]:
```suggestion
import VitessceSlide320w from 'portal-images/vitessce-slide-320w.png';
import VitessceSlide640w from 'portal-images/vitessce-slide-640w.png';
import VitessceSlide1280w from 'portal-images/vitessce-slide-1280w.png';
import VitessceSlide1392w from 'portal-images/vitessce-slide-1392w.png';
import AzimuthSlide320w from 'portal-images/azimuth-slide-320w.png';
import AzimuthSlide640w from 'portal-images/azimuth-slide-640w.png';
import AzimuthSlide1280w from 'portal-images/azimuth-slide-1280w.png';
import AzimuthSlide1392w from 'portal-images/azimuth-slide-1392w.png';
import CCFSlide320w from 'portal-images/ccf-slide-320w.png';
import CCFSlide640w from 'portal-images/ccf-slide-640w.png';
import CCFSlide1280w from 'portal-images/ccf-slide-1280w.png';
import CCFSlide1392w from 'portal-images/ccf-slide-1392w.png';
```

**Review by mccalluc — COMMENTED** [review]
Looks good:
- I think the mod is a tad simpler, if that's ok with you.
- Fewer images: 320/640/1280 as a geometric series, and 1392, since we have it.

I made PR in the subrepo, but that doesn't need to block this. Thanks for pulling out the images.

**Pull request merged**